### PR TITLE
feat: add default filter types

### PR DESCRIPTION
**Pull request by magrinj**
This PR fix issue #1900.
We're mirroring all default filter types of pg_graphql in order to add `where` conditions in our dynamic mutations and queries.

### CODE REVIEW

File: server/src/tenant/schema-builder/graphql-types/input/filter-is-enum-filter.type.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { GraphQLEnumType } from 'graphql';
+
+export const FilterIsEnumType = new GraphQLEnumType({
+  name: 'FilterIs',
+  values: {
+    PENDING: { value: 'PENDING' },
```
Comment (Weiko) [inline server/src/tenant/schema-builder/graphql-types/input/filter-is-enum-filter.type.ts:6]: Are those test values? Should we include this filter in this PR?

File: server/src/tenant/schema-builder/graphql-types/input/big-float-filter.type.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLFloat,
+} from 'graphql';
+
+import { FilterIsEnumType } from './filter-is-enum-filter.type';
+
+export const BigFloatFilterType = new GraphQLInputObjectType({
+  name: 'BigFloatType',
+  fields: () => ({
```
Comment (Weiko) [inline server/src/tenant/schema-builder/graphql-types/input/big-float-filter.type.ts:12]: It seems fields is not instantiate the same way comparing to other filters, is it intentional?

File: server/src/tenant/schema-builder/utils/map-column-type-to-filter-type.util.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Map the column type from field-metadata to its corresponding filter type.
+ * @param columnType Type of the column in the database.
+ */
+export const mapColumnTypeToFilterType = (column: FieldMetadata) => {
+  switch (column.type) {
+    case 'uuid':
+      return UUIDFilterType;
+    case 'text':
```
Comment (Weiko) [inline server/src/tenant/schema-builder/utils/map-column-type-to-filter-type.util.ts:19]: To discuss: It seems we are doing this kind of check in many places now, we should definitely see if we can have something cleaner to implement new types without editing many files. (An interface class with a class for each type where we need to implement some methods for example?)

**Review by Weiko — APPROVED** [review]
LGTM, left some comments @magrinj

File: server/src/tenant/schema-builder/graphql-types/input/filter-is-enum-filter.type.ts (hunk 1 above)
Comment (magrinj) [reply server/src/tenant/schema-builder/graphql-types/input/filter-is-enum-filter.type.ts:6]: @Weiko Don't really know the goal of that for now, but it's in what `pg_graphql` is generating:
https://supabase.github.io/pg_graphql/example_schema/

File: server/src/tenant/schema-builder/graphql-types/input/big-float-filter.type.ts (hunk 2 above)
Comment (magrinj) [reply server/src/tenant/schema-builder/graphql-types/input/big-float-filter.type.ts:12]: No you're right, it doesn't need to be a function, I'll make if more clean

File: server/src/tenant/schema-builder/utils/map-column-type-to-filter-type.util.ts (hunk 3 above)
Comment (magrinj) [reply server/src/tenant/schema-builder/utils/map-column-type-to-filter-type.util.ts:19]: Yes it's planed but I'm waiting for now until everything is working smoothly, and I'll make something more clean :)

**Pull request merged**